File: spigot/src/main/java/net/azisaba/azisabaachievements/spigot/commands/CommandCreate.java
package net.azisaba.azisabaachievements.spigot.commands;

import net.azisaba.azisabaachievements.api.AzisabaAchievementsProvider;
import net.azisaba.azisabaachievements.api.Key;
import net.azisaba.azisabaachievements.spigot.command.Command;
import org.bukkit.ChatColor;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

public class CommandCreate implements Command {
    @Override
    public void execute(@NotNull CommandSender sender, @NotNull String @NotNull [] args) {
        if (args.length < 3) {
            sender.sendMessage(ChatColor.RED + "Usage: " + getFullUsage());
            return;
        }
        Key key = Key.key(args[0]);
        long count = Long.parseLong(args[1]);
        int point = Integer.parseInt(args[2]);
        AzisabaAchievementsProvider.get().getAchievementManager()
                .createAchievement(key, count, point)
                .whenComplete((data, throwable) -> {
                    if (throwable != null) {
                        sender.sendMessage(ChatColor.RED + "Failed to create achievement: " + throwable.getMessage());
                        return;
                    }
                    sender.sendMessage(ChatColor.GREEN + "Successfully created achievement: " + data.getKey());
                });
    }

    @Override
    public @NotNull String getName() {
        return "create";
    }

    @Override
    public @NotNull String getDescription() {
        return "Creates a new achievement.";
    }

    @Override
    public @NotNull String getUsage() {
        return "<key> <count> <point>";
    }
}

File: cli/src/main/kotlin/net/azisaba/azisabaachievements/cli/commands/CommandCreate.kt
package net.azisaba.azisabaachievements.cli.commands

import kotlinx.cli.ArgType
import kotlinx.cli.ExperimentalCli
import kotlinx.cli.Subcommand
import kotlinx.cli.required
import net.azisaba.azisabaachievements.api.Key
import net.azisaba.azisabaachievements.cli.CLIMain
import net.azisaba.azisabaachievements.cli.Colors
import net.azisaba.azisabaachievements.cli.ExtraArgType
import net.azisaba.azisabaachievements.cli.Log
import net.azisaba.azisabaachievements.cli.achievement.CLIAchievementManager

@OptIn(ExperimentalCli::class)
object CommandCreate : Subcommand("create", "Creates a new achievement") {
    private val key by option(ArgType.String, shortName = "k", description = "The key of the achievement").required()
    private val count by option(ExtraArgType.Long, shortName = "c", description = "The number of progress required to unlock the achievement").required()
    private val point by option(ArgType.Int, shortName = "p", description = "The point of the achievement").required()

    override fun execute() {
        CLIMain.init()
        val keyKey = Key.key(key)
        CLIAchievementManager.createAchievement(keyKey, count, point)
            .whenComplete { data, throwable ->
                if (throwable != null) {
                    Log.error("Failed to create achievement: ${throwable.message}")
                    return@whenComplete
                }
                Log.info("Created achievement: ${Colors.YELLOW}${data.key}${Colors.RESET} (ID: ${Colors.YELLOW}${data.id})")
            }
    }
}
